### PR TITLE
[FEATURE]: Handle creating a service file for croncatd

### PR DESCRIPTION
**Pull request by SeedyROM**
**Create feature that was mentioned in #32.**
- [x] Change verbiage for `daemon` to `setup-service`, to make it slightly more descriptive of what's going on?
- [x] Use the command `croncatd --chain-id <chain-id> setup-service` to create system services in a relative folder called `system-services`.
    - For instance `croncatd --chain-id local setup-service` creates a file at path `./system-services/croncatd-local.service`.
- [x] Fix strange use of croncatd chain_id being optional but also not optional?
- [x] Refactor system into it's own sub module folder.
- [x] Define the `ServiceDaemon` struct to help build system service files.
- [x] Link the newly created service into the system via a sudo command.

## Resources:
- Calling sudo and capturing stdin: https://users.rust-lang.org/t/how-to-execute-a-root-command-on-linux/50066/8

### CODE REVIEW

**Comment by SeedyROM** [comment]
Idea of note (handling only systemd?): https://github.com/CronCats/croncat-rs/issues/32#issuecomment-1253461479

**Comment by SeedyROM** [comment]
ProTip: Don't use github mobile to try and link PRs with issues, :'( 

Sorry for the spam!

**Comment by mikedotexe** [comment]
Thanks @SeedyROM I am all in favor of your suggestion of `create-service`
Feel free to hit the 👍🏼 or 👎🏼 emoji on it here

**Comment by SeedyROM** [comment]
@mikedotexe @TrevorJTClarke I went with `setup-service`, but this is not a huge change in the future. Any thoughts?

File: croncatd/banner.txt [hunk 1]
```diff
@@ -3,4 +3,4 @@
   /\_/\   / __| '__/ _ \| '_ \ / __/ _` | __/ _` |
  ( o.o ) | (__| | | (_) | | | | (_| (_| | || (_| |
   > ^ <   \___|_|  \___/|_| |_|\___\__,_|\__\__,_|
-==================================================
+===================================================
```
Comment (TrevorJTClarke) [inline croncatd/banner.txt:6]: This.
Normalize comment incrementing

**Review by TrevorJTClarke — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -30,13 +30,33 @@ croncatd 0.1.0
 The croncat agent daemon.
 
 USAGE:
-    croncatd [FLAGS]
+    croncatd [FLAGS] --chain-id <chain-id> <SUBCOMMAND>
```
Comment (Buckram123) [inline README.md:33]: Thanks for updating the readme <3

File: croncat/src/system/service.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    [Install]
+                    WantedBy=multi-user.target
+                "},
+                chain_id = chain_id,
+                user = user,
+                service_dir = full_service_dir_path.to_str().ok_or_else(|| eyre!(
+                    "Could not convert daemon service directory path to string",
+                ))?,
+                exe_path = std::env::current_exe()?.to_str().ok_or_else(|| eyre!(
+                    "Could not convert daemon service executable path to string",
```
Comment (Buckram123) [inline croncat/src/system/service.rs:92]: that looks so great

File: croncat/src/store/mod.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+//! Helpers to handle storing and retrieving data from the local filesystem.
+
```
Comment (Buckram123) [inline croncat/src/store/mod.rs:2]: 🔥

File: croncat/src/config/mod.rs [hunk 5]
```diff
@@ -23,13 +24,9 @@ impl ChainConfig {
...
-    pub async fn new(chain_id: Option<&str>) -> Result<Self, Report> {
-        let mut ch_id=chain_id.unwrap();
-        if !chain_id.is_some() {
-            ch_id = "uni-3";
-        }
-        let config_file = &format!("config.{}.yaml", ch_id);
-        let config_override_file = &format!("config.{}.override.yaml", ch_id);
+    pub async fn new(chain_id: &String) -> Result<Self, Report> {
+        let config_file = &format!("config.{}.yaml", chain_id);
+        let config_override_file = &format!("config.{}.override.yaml", chain_id);
```
Comment (Buckram123) [inline croncat/src/config/mod.rs:29]: thanks for fixing it

**Review by Buckram123 — APPROVED** [review]
Great job! 🧨🧨🧨
And thanks for updating the `README`!

**Comment by Buckram123** [comment]
Funny it doesn't count my approve, cause I dont have an right access to this repo

**Comment by SeedyROM** [comment]
> Funny it doesn't count my approve, cause I dont have an right access to this repo

It counts for me hahaha.

File: croncatd/banner.txt (hunk 1 above)
Comment (mikedotexe) [reply croncatd/banner.txt:6]: Normalize adding some equality to this world.

**Pull request merged**